utils/integral-tests/src/runtime.rs

```diff
@@ -0,0 +1,965 @@
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use frame_election_provider_support::{
+    onchain, ElectionDataProvider, NposSolution, SequentialPhragmen, VoteWeight,
+};
+pub use frame_support::{
+    construct_runtime,
+    dispatch::{DispatchClass, WeighData},
+    parameter_types,
+    traits::{
+        ConstU128, ConstU16, ConstU32, Contains, Currency, EitherOfDiverse, EqualPrivilegeOnly,
+        Everything, FindAuthor, KeyOwnerProofSystem, LockIdentifier, OnUnbalanced, Randomness,
+        StorageInfo, U128CurrencyToVote,
+    },
+    weights::{
+        constants::{
+            BlockExecutionWeight, ExtrinsicBaseWeight, RocksDbWeight, WEIGHT_PER_MILLIS,
+            WEIGHT_PER_SECOND,
+        },
+        IdentityFee, Weight,
+    },
+    PalletId, StorageValue,
+};
+use frame_system::{EnsureRoot, EnsureSigned};
+use pallet_election_provider_multi_phase::SolutionAccuracyOf;
+use pallet_grandpa::AuthorityId as GrandpaId;
+use pallet_im_online::sr25519::AuthorityId as ImOnlineId;
+use pallet_session::historical::{self as pallet_session_historical};
+pub use pallet_timestamp::Call as TimestampCall;
+pub use pallet_transaction_payment::{CurrencyAdapter, Multiplier};
+pub use runtime_primitives::{AccountId, Signature};
+use runtime_primitives::{Balance, BlockNumber, Hash, Index, Moment};
+use sp_core::crypto::KeyTypeId;
+use sp_runtime::{
+    curve::PiecewiseLinear,
+    generic, impl_opaque_keys,
+    traits::{AccountIdLookup, BlakeTwo256, OpaqueKeys},
+    transaction_validity::TransactionPriority,
+    FixedU128, Perbill, Percent, Permill,
+};
+use sp_std::{
+    convert::{TryFrom, TryInto},
+    prelude::*,
+};
+use static_assertions::const_assert;
+
+pub use frame_system::Call as SystemCall;
+pub use pallet_balances::Call as BalancesCall;
+pub use pallet_staking::StakerStatus;
+pub use pallet_sudo::Call as SudoCall;
+pub use sp_runtime::BuildStorage;
+
+pub mod constants {
+    /// Currency related constants
+    pub mod currency {
+        use runtime_primitives::Balance;
+
+        /// The existential deposit.
+        pub const EXISTENTIAL_DEPOSIT: Balance = MILLICENTS / 10; // 1_000
+
+        // TODO: review quantities based on economic model (issue #1277)
+        pub const UNITS: Balance = 1_000_000_000; // 10^(-9) precision
```

Comment:
Was it expected to use ^9 here?

Reply:
No reason for having smaller precision, just dealing with smaller numbers. I suggest to keep it this way for now to avoid risk of tests failing in case we align it with 12 decimals as we have in Vara.